src/main/java/frc/robot/subsystems/DriveSubsystem.java

```diff
@@ -84,7 +84,7 @@ public boolean gyroDrive(double endPositionTick, double heading, double power) {
         System.out.println("error: " + error + "  leftPower: " + adjustedLeftPower + " rightPower:" + adjustedRightPower + "  distanceToTarget: " + distanceToTarget);
 		//the scalar slows down the robot as it gets closer to the goal,
 		//while ensuring that it doesn't slow the robot down too much (keeps scalar between 1 and 0.1)
-		double scalar = 1; //Math.min(1, Math.max(Math.atan(distanceToTarget / 4) * 2 / Math.PI, 0.1));
+		double scalar = Math.min(1, Math.max(distanceToTarget / 24, 0.05)); //Math.min(1, Math.max(Math.atan(distanceToTarget / 4) * 2 / Math.PI, 0.1));
```

Comment:
Right now we're calculating and bounding these two variables power and scalar independently, and combining them at the end. This doesn't have to be so complicated.

Think about what basic mathematical relation between power and error is. We want power=constant*error, then bound power between upper and lower limits. If it's easier, you can calculate the magnitude of the power and apply the bounds first, then set the right sign for direction.